src/Integration.Vsix/SonarLintTagger/AnalysisScheduler.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    [Export(typeof(IAnalysisScheduler))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class AnalysisScheduler : IAnalysisScheduler
+    {
+        private readonly ILogger logger;
+        private readonly IDictionary<string, CancellationTokenSource> analysisJobs;
```

Comment:
Out of interest, did you consider using a thread-safe `ConcurrentDictionary` instead a normal dictionary with locking?

Reply:
Since I needed two actions: Get (to dispose the previous) and Update (to set the new), and there is no concurrent option for this, so I'd have to use a lock anyway.